datafusion/core/src/logical_plan/builder.rs

```diff
@@ -1055,19 +1055,23 @@ pub fn union_with_alias(
     right_plan: LogicalPlan,
     alias: Option<String>,
 ) -> Result<LogicalPlan> {
-    let inputs = vec![left_plan, right_plan]
+    let union_schema = left_plan.schema().clone();
+    let inputs_iter = vec![left_plan, right_plan]
```

Comment:
I wonder if there is a reason you needed to make changes below this line (other than to use `union_schema` rather than `schema` when creating the `Projection`

In particular, I wonder if we can avoid the extra call to `clone()` by going back to the previous setup?